Update In-Place Pod Resize docs for v1.32 (#48503)
kubernetes/website: content/en/docs/tasks/configure-pod-container/resize-container-resources.md

[workload controller](/docs/concepts/workloads/controllers/) make a replacement Pod
that has a different resource requirement.

> A resize request is made through the pod `/resize` subresource, which takes the full updated pod for

Reviewer: nit: use Pod, not pod throughout this diff.

> an update request, or a patch on the pod object for a patch request.

Reviewer: ```suggestion
an update request, or a patch on the `Pod` object for a patch request.
```

For in-place resize of pod resources:
- A container's resource `requests` and `limits` are _mutable_ for CPU
  and memory resources. These fields represent the _desired_ resources for the container.

> - The `resources` field in `containerStatuses` of the Pod's status reflects the resources
>   _allocated_ to the pod's containers. For running containers, this reflects the actual resource
>   `requests` and `limits` that are configured as reported by the container runtime. For non-running
>   containers, these are the resources allocated for the container when it starts.

Reviewer: ```suggestion
- The `containerStatuses.resources` field in the Pod `status` field reflects the resources
   that are _allocated_ to the Pod's containers. For running containers, these are the actual values
   of `requests` and `limits`, as reported by the container runtime. For non-running containers,
   these are the resources that are allocated for when the container starts.
```

- The `resize` field in the Pod's status shows the status of the last requested
  pending resize. It can have the following values:

>   - `Proposed`: This value indicates that a pod was resized, but the Kubelet has not yet processed
>     the resize.

Reviewer: ```suggestion
  - `Proposed`: this value indicates that a Pod resize request was received, but the
     kubelet has not yet processed the resize.
```

  - `InProgress`: This value indicates that the node has accepted the resize
    request and is in the process of applying it to the pod's containers.
  - `Deferred`: This value means that the requested resize cannot be granted at
    this time, and the node will keep retrying. The resize may be granted when

>     other pods are removed and free up node resources.

Reviewer: ```suggestion
    other pods are removed and the available node resource capacity increases.
```

  - `Infeasible`: is a signal that the node cannot accommodate the requested
    resize. This can happen if the requested resize exceeds the maximum
    resources the node can ever allocate for a pod.

>   - `""`: An empty or unset value indicates that the last resize completed. This should only be the
>     case if the resources in the container spec match the resources in the container status.

Reviewer: ```suggestion
  - `""`: An empty or unset value indicates that the most recent pending resize was completed.
     An empty or unset value only appears if the resources in the container specification match
     the resources in the container status.
```

> If a node has pods with an incomplete resize, the scheduler will compute the pod requests from the
> maximum of a container's desired resource requests, and it's actual requests reported in the status.

Reviewer: ```suggestion
If a node has Pods with an incomplete resize, the scheduler uses the largest of the following values to
calculate the Pod requests:

*  The desired resource requests for all containers in the Pod, as specified in the
   `spec.containers.resources.requests` field.
*  The actual resource requests for all containers in the Pod, as reported in the
   `spec.status.containerStatuses.resources.requests` field.
```

[workload controller](/docs/concepts/workloads/controllers/) make a replacement Pod
that has a different resource requirement.

A resize request is made through the pod `/resize` subresource, which takes the full updated pod for
an update request, or a patch on the pod object for a patch request.

For in-place resize of pod resources:
- A container's resource `requests` and `limits` are _mutable_ for CPU
  and memory resources. These fields represent the _desired_ resources for the container.
- The `resources` field in `containerStatuses` of the Pod's status reflects the resources
  _allocated_ to the pod's containers. For running containers, this reflects the actual resource
  `requests` and `limits` that are configured as reported by the container runtime. For non-running
  containers, these are the resources allocated for the container when it starts.

> - The `resize` field in the Pod's status shows the status of the last requested

Reviewer 2: Do you think, we should add a sentence when `resize` value is null ? For example if a pod was resized , and pending resize is null means that Kubelet has processed the resize ?

<!-- steps -->

> ## Limitations

Reviewer: This belongs further up. Make this either a subheading of the prerequisites heading or a main heading before the prerequisites heading.

> In-place resize of pod resources currently has the following limitations:

Reviewer: ```suggestion
In-place resize of Pod resources has the following limitations:
```

- Only CPU and memory resources can be changed.

> - Pod QoS Class cannot change. This means that requests must continue to equal limits for Guaranteed
>   pods, Burstable pods cannot set requests and limits to be equal for both CPU & memory, and you
>   cannot add resource requirements to Best Effort pods.

Reviewer: ```suggestion
- The Pod QoS Class cannot change. The following considerations continue to apply:
   - Guaranteed QoS Pods: requests must be equal to limits.
   - Burstable QoS Pods: requests can't be equal to limits for both CPU and memory
   - Best Effort QoS Pods:  you can't set resource requirements
```

> - Init containers and Ephemeral Containers cannot be resized.

Reviewer: ```suggestion
- Init containers and ephemeral containers cannot be resized.
```

- Resource requests and limits cannot be removed once set.

> - A container's memory limit may not be reduced below its usage. If a request puts a container in
>   this state, the resize status will remain in `InProgress` until the desired memory limit becomes
>   feasible.

Reviewer: ```suggestion
- A container's memory limit can't be reduced below the memory usage. If a request puts a 
  container in this state, the resize will remain in the `InProgress` state until the desired
  memory limit becomes  feasible.
```

    started: true
```

Observe that the `resources` in the `containerStatuses` have been updated to reflect the new desired

> CPU requests. This indicates that node was able to accommodate the increased CPU resource needs.c,

Reviewer 2: Unnecessary `.c`

Author: Thanks, fixed.
